ntbs-service/DataAccess/UserRepository.cs

```diff
@@ -65,13 +66,13 @@ public async Task<User> GetUserByUsername(string username)
         /// </summary>
         public async Task UpdateUserContactDetails(User user)
         {
-            _context.Attach(user);
-            _context.Entry(user).Property(x => x.JobTitle).IsModified = true;
-            _context.Entry(user).Property(x => x.PhoneNumberPrimary).IsModified = true;
-            _context.Entry(user).Property(x => x.PhoneNumberSecondary).IsModified = true;
-            _context.Entry(user).Property(x => x.EmailPrimary).IsModified = true;
-            _context.Entry(user).Property(x => x.EmailSecondary).IsModified = true;
-            _context.Entry(user).Property(x => x.Notes).IsModified = true;
+            var existingUser = await _context.User.SingleAsync(u => u.Username == user.Username);
+            existingUser.JobTitle = user.JobTitle;
+            existingUser.PhoneNumberPrimary = user.PhoneNumberPrimary;
+            existingUser.PhoneNumberSecondary = user.PhoneNumberSecondary;
+            existingUser.EmailPrimary = user.EmailPrimary;
+            existingUser.EmailSecondary = user.EmailSecondary;
+            existingUser.Notes = user.Notes;

```

Comment:
The previous way of doing this is quite fragile, because it relies on the fact that the user has not already been brought into the context. This will achieve the same result, whether or not the user is already in the context.